src/FsAutoComplete.Core/Commands.fs

```diff
@@ -74,17 +74,17 @@ module AsyncResult =
   let recoverCancellation (ar: Async<Result<CoreResponse<'t>, exn>>) =
     recoverCancellationGeneric ar (sprintf "Request cancelled (exn was %A)" >> CoreResponse.InfoRes)
 
-  let recoverCancellationIgnore (ar: Async<Result<unit, exn>>) = AsyncResult.foldResult id ignore ar
+  let recoverCancellationIgnore (ar: Async<Result<unit, exn>>) = ar |> AsyncResult.foldResult id (ignore<exn>)
 
 [<RequireQualifiedAccess>]
 type NotificationEvent =
-  | ParseError of errors: FSharpDiagnostic[] * file: string<LocalPath>
+  | ParseError of errors: FSharpDiagnostic[] * file: string<LocalPath> * version: int
```

Comment:
I really wonder if we need some kind of 'file@version' abstraction here. Not now, but just immediately comes to mind.